web-marketplace/src/components/organisms/UserAccountSettings/UserAccountSettings.tsx

```diff
@@ -0,0 +1,116 @@
+            >
+              <ConnectField {...provider} />
+            </div>
+          ))}
+        </div>
+      </div>
+      <div className="flex flex-col gap-30">
+        <div className="flex flex-col gap-10">
+          <Subtitle>Wallet integration</Subtitle>
+          <Body size="sm" color="info.dark-grey">
+            Connect a wallet address to be able to perform on chain tasks such
+            as buying and selling credits, and creating projects on Regen
+            Ledger.
+            {/* TODO: add link or button to learn more when we have the content ready */}
+            <a>Learn more</a>
+          </Body>
+        </div>
+        <ConnectField
+          providerName="Keplr or Wallet Connect"
+          {...walletProvider}
+        />
+      </div>
+    </div>
+  );
+};
+
+const ConnectField = ({
+  providerName,
+  connect,
+  disconnect,
+  address,
+}: {
+  providerName: string;
+  connect?: () => void;
+  disconnect?: () => void;
+  address?: string;
+}) => {
+  return (
+    <div className="flex flex-row justify-between">
+      <div className="flex flex-col gap-5">
+        <Body size="md" color="info.dark-grey">
+          {providerName}
+        </Body>
+        {address ? (
+          <AddressWidget address={address} />
+        ) : (
+          <Body size="sm">{connect ? 'Disconnected' : 'Connected'}</Body>
+        )}
+      </div>
+      <div>
+        {connect ? (
+          <ContainedButton onClick={connect}>CONNECT</ContainedButton>
+        ) : (
+          <OutlinedButton onClick={disconnect}>
+            <div className="flex flex-row gap-10 items-center">
+              {/* the SVG source for this icon is included inline so that styling with tailwind works */}
+              <svg className="stroke-secondary-main" width="24" height="24" fill="none">
+                <rect width="2" height="17.958" x="6.414" y="19.113" fill="#545555" rx=".25" transform="rotate(-135 6.414 19.113)"/><rect width="2" height="17.958" fill="#545555" rx=".25" transform="scale(1 -1) rotate(45 32.021 11.83)"/>
+              </svg>
+              <div>DISCONNECT</div>
+            </div>
+          </OutlinedButton>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const AddressWidget = ({ address }: { address: string }) => (
+  <div className="flex flex-row gap-5">
+    <Body size="sm">{address}</Body>
+    <svg width="24" height="24" fill="none"
+         onClick={() => {
+           navigator.clipboard.writeText(address);
+         }}
+         className="cursor-pointer hover:stroke-grey-300"
+    >
+      <path fill="#000" fill-rule="evenodd" d="M19 8v9H9.25a.25.25 0 0 1-.25-.25V3h5v4.25c0 .414.336.75.75.75H19Zm-4-4.586L18.287 7H15V3.414ZM3.25 5H7V1.25A.25.25 0 0 1 7.25 1h8.14c.07 0 .137.03.184.081l5.36 5.847a.25.25 0 0 1 .066.17V18.75a.25.25 0 0 1-.25.25H17v3.75a.25.25 0 0 1-.25.25H3.25a.25.25 0 0 1-.25-.25V5.25A.25.25 0 0 1 3.25 5ZM7 16.75V7H5v14h10v-2H7.25a.25.25 0 0 1-.25-.25v-2Z" clip-rule="evenodd"/>
+    </svg>
```

Comment:
Note the inline svg here and for the x icon above. This is necessary for tailwind styling. Should I pull these out into standalone components in web-components maybe?

Reply:
Also for this copy button, I would prefer to use a headless button component rather than an on click handler for an svg element. Doesn't need to be in this PR necessarily, but I think it would be a good practice. React Aria Components has one: https://react-spectrum.adobe.com/react-aria/Button.html. @flagrede I know you expressed concern about react aria components being in alpha, now it's in beta and to me feels like a pretty solid library overall. If Radix Primitives had a button component, I would happily use that instead, but they don't (https://github.com/radix-ui/primitives/issues/892 - note the involvement of the React Aria maintainer here).

Reply:
> Note the inline svg here and for the x icon above. This is necessary for tailwind styling. Should I pull these out into standalone components in web-components maybe?

This is usually what we do for icons, see web-components/src/components/icons, although they all rely on MUI `SvgIcon` currently
I believe we already had some x icon btw, see "CloseIcon" in https://regen-storybook.netlify.app/?path=/story/icons--all-icons but I'm not sure if that's exactly the same here cc/ @erikalogie 
It also looks a bit darker than on figma

Reply:
The close x button is the same everywhere (but different colors though obviously)

Reply:
If it's in beta now it should be safer to use. We're not taking too much with a button component anyway, my concern was more for more advanced components. I like Radix UI overall but react-aria might be a better fit for us, let's start with that.